homeassistant/components/songpal/media_player.py

```diff
@@ -72,46 +62,23 @@ async def async_setup_entry(
     hass: HomeAssistantType, config_entry: ConfigEntry, async_add_entities
 ) -> None:
     """Set up songpal media player."""
-    if DOMAIN not in hass.data:
-        hass.data[DOMAIN] = {}
-
     name = config_entry.data[CONF_NAME]
     endpoint = config_entry.data[CONF_ENDPOINT]
 
-    if endpoint in hass.data[DOMAIN]:
-        _LOGGER.debug("The endpoint exists already, skipping setup.")
-        return
-
     device = SongpalDevice(name, endpoint)
     try:
         await device.initialize()
```

Comment:
Just a suggestion for the future: It would be more clear if we separated the device api from the entity api. Maybe make a new wrapper class for the device and instantiate this before creating the entity and pass that to the entity.